develop/migration_mt_django.rst

```diff
@@ -88,84 +197,322 @@ Django automatically creates a simple "id" primary key on models, so we will nee
+
+        ALTER TABLE myapp_task
+        ADD CONSTRAINT myapp_task_pkey
+        PRIMARY KEY (account_id, id)
+      """),
+
+      migrations.RunSQL("""
+        ALTER TABLE myapp_projectmanager
+        DROP CONSTRAINT myapp_projectmanager_pkey CASCADE;
+
+        ALTER TABLE myapp_projectmanager
+        ADD CONSTRAINT myapp_projectmanager_pkey PRIMARY KEY (account_id, id);
       """),
     ]
 
-**3. Switch to TenantModel**
 
-Next, we'll use the `django-multitenant <https://github.com/citusdata/django-multitenant>`_ library to add store_id to foreign keys, and make application queries easier later on.
+**2.2 Including the account\_id to unique constraints**
+
+The same thing needs to be done for ``UNIQUE`` constraints. You can have explicit constraints that you might have set in your model with ``unique=True`` or ``unique_together`` like:
+
+
+.. code-block:: python
+
+  class Project(models.Model):
+      name = models.CharField(max_length=255, unique=True)
+      account = models.ForeignKey(Account, related_name='projects',
+                                  on_delete=models.CASCADE)
+      managers = models.ManyToManyField(Manager, through='ProjectManager')
+
+
+  class Task(models.Model):
+      name = models.CharField(max_length=255)
+      project = models.ForeignKey(Project, on_delete=models.CASCADE,
+                                  related_name='tasks')
+      account = models.ForeignKey(Account, related_name='tasks',
+                                  on_delete=models.CASCADE)
+
+      class Meta:
+          unique_together = [('name', 'project')]
+
+
+
+For these constraints, you can simply change in the models the constraints:
+
+.. code-block:: python
+
+  class Project(models.Model):
+      name = models.CharField(max_length=255)
+      account = models.ForeignKey(Account, related_name='projects',
+                                  on_delete=models.CASCADE)
+      managers = models.ManyToManyField(Manager, through='ProjectManager')
+
+      class Meta:
+          unique_together = [('account', 'name')]
+
+
+
+  class Task(models.Model):
+      name = models.CharField(max_length=255)
+      project = models.ForeignKey(Project, on_delete=models.CASCADE,
+                                  related_name='tasks')
+      account = models.ForeignKey(Account, related_name='tasks',
+                                  on_delete=models.CASCADE)
+
+      class Meta:
+          unique_together = [('account', 'name', 'project')]
+
+
+Then generate the migration with:
+
+.. code-block:: python
+
+  python manage.py makemigrations
+
+
+Some ``UNIQUE`` constraints are created by the ORM and you will need to explicitily drop them.
+This is the case for ``OneToOneField`` and ``ManyToMany`` fields.
+
+
+For these cases you will need to:
+1. Find the constraints
+2. Do a migration to drop them
+3. Re-create constraints including the account\_id field
+
+To find the constraints, connect to your database and run ``\d+ myapp_projectmanager``
```

Comment:
Might want to note that we're asking them to connect with psql in particular. (Django has its own command line db console.)